src/Common/src/Interop/Unix/System.Net.Security.Native/SecuritySafeHandles.cs

```diff
@@ -13,16 +13,68 @@ namespace System.Net.Security
     internal sealed class SafeFreeNegoCredentials : SafeFreeCredentials
     {
         private SafeGssCredHandle _credential;
+        private readonly bool _isNtlm;
+        private readonly string _userName;
+        private readonly bool _isdefault;
 
         public SafeGssCredHandle GssCredential
         {
             get { return _credential; }
         }
 
-        public SafeFreeNegoCredentials(string username, string password, string domain) : base(IntPtr.Zero, true)
+        public bool IsNtlm
         {
+            get { return _isNtlm; }
+        }
+
+        public string UserName
+        {
+            get { return _userName; }
+        }
+
+        public bool IsDefault
+        {
+            get { return _isdefault; }
+        }
+
+        public SafeFreeNegoCredentials(bool ntlmOnly, string username, string password, string domain)
+            : base(IntPtr.Zero, true)
+        {
+            Debug.Assert(username != null && password != null, "Username and Password can not be null");
+            const char at = '@';
+            const char backwhack = '\\';
+
+            if (username.IndexOf(backwhack) > 0 && string.IsNullOrEmpty(domain))
+            {
+                string[] nameArray = username.Split(backwhack);
```

Comment:
Rather than using Split, can we use the index found above to do the right Substring calls and avoid the extra array allocation?  You could do another IndexOf starting after the found location to see if there's another occurrence, too.